File: problems/Add Strings/addStrings.java
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuilder sb = new StringBuilder();
        int carry = 0;
        int idx1 = num1.length() - 1;
        int idx2 = num2.length() - 1;

        while (0 < carry || 0 <= idx1 || 0 <= idx2) {
            int num = carry;
            num += (0 <= idx1) ? (num1.charAt(idx1--) - '0') : 0;
            num += (0 <= idx2) ? (num2.charAt(idx2--) - '0') : 0;

            sb.append(num%10);
            carry = num/10;
        }

        return sb.reverse().toString();
    }
}

File: problems/Add Strings/addStrings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
    const result = [];
    const zeroCode = '0'.charCodeAt();
    let idx1 = num1.length - 1;
    let idx2 = num2.length - 1;
    let carry = 0;

    while (0 <= idx1 || 0 <= idx2 || 0 !== carry) {
        let sum = carry;

        if (0 <= idx1) {
            sum += num1.charCodeAt(idx1--) - zeroCode;
        }

        if (0 <= idx2) {
            sum += num2.charCodeAt(idx2--) - zeroCode;
        }

        result.push(sum % 10);
        carry = Math.floor(sum / 10);
    }

    result.reverse();

    return result.join('');
};
